vertex_ai_dataset/main.tf

```
@@ -0,0 +1,51 @@
+# [START vertex_ai_enable_api]
+resource "google_project_service" "aiplatform" {
+  provider           = google-beta
+  service            = "aiplatform.googleapis.com"
+  disable_on_destroy = false
+}
+# [END vertex_ai_enable_api]
+
+# [START vertex_ai_dataset_image]
+resource "google_vertex_ai_dataset" "image-dataset" {
```

nit: prefer underscores for resource names https://cloud.google.com/docs/terraform/best-practices-for-terraform#naming-convention here and throughout.
```suggestion
resource "google_vertex_ai_dataset" "image_dataset" {
```